File: sdk-python/example_agent.py
import os
from client import AgentWallet
from dotenv import load_dotenv
import requests

load_dotenv()

def main():
    print("🤖 X402 AgentWallet Python Demo")
    print("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━")

    agent = AgentWallet(verbose=True)

    print(f"Wallet agente: {agent.address}")
    print(f"Saldo USDC:    {agent.balance():.2f} USDC")
    print("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━")

    # Preview gratuita
    print("\n📄 Chiamata preview gratuita...")
    preview = requests.get("http://127.0.0.1:5001/data/preview",verify=False)
    print("Preview status:", preview.status_code)
    print("Preview text:", preview.text[:200])
    import json
    print("Preview:", json.dumps(preview.json(), indent=2, ensure_ascii=False))

    # Dataset completo — paga automaticamente
    print("\n💎 Chiamata dataset completo (x402)...")
    try:
        data = agent.fetch("http://127.0.0.1:5001/data/full")
        print("\n✅ Dataset completo ricevuto!")
        print("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━")
        print(json.dumps(data, indent=2, ensure_ascii=False))
    except Exception as e:
        print(f"\n❌ Errore: {e}")

if __name__ == '__main__':
    main()

File: sdk-node/example-agent.js
require("dotenv").config();
const { AgentWallet } = require("./client");

async function main() {
  console.log("🤖 X402 AgentWallet Demo");
  console.log("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");

  const agent = new AgentWallet({
    privateKey: process.env.X402_AGENT_PRIVATE_KEY,
    verbose:    true,
  });

  console.log("Wallet agente:", agent.address);

  // Controlla saldo USDC
  const balance = await agent.balance();
  console.log("Saldo USDC:   ", balance, "USDC");
  console.log("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");

  try {
    // Prima chiama la preview gratuita
    console.log("\n📄 Chiamata preview gratuita...");
    const fetch = (...args) => import("node-fetch").then(({ default: f }) => f(...args));
    const preview = await fetch("http://localhost:4000/data/preview");
    const previewData = await preview.json();
    console.log("Preview:", JSON.stringify(previewData, null, 2));

    // Poi chiama il dataset completo — paga automaticamente se necessario
    console.log("\n💎 Chiamata dataset completo (x402)...");
    const data = await agent.fetch("http://localhost:4000/data/full");

    console.log("\n✅ Dataset completo ricevuto!");
    console.log("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");
    console.log(JSON.stringify(data, null, 2));

  } catch (err) {
    console.error("\n❌ Errore:", err.message);
  }
}

main().catch(console.error);
